### PR TITLE
Adding comment to describe the logic for blob content validation

### PR DESCRIPTION
**Pull request by soninaren**
Resolves - [Link](https://github.com/orgs/Azure/projects/473/views/9?sliceBy%5Bvalue%5D=surgupta-msft&pane=issue&itemId=80981198)

Guidance for the changes is shared in Codeql links from Engineering hub - [Link1](https://eng.ms/docs/cloud-ai-platform/devdiv/one-engineering-system-1es/1es-docs/codeql/codeql-semmle), [Link2](https://eng.ms/docs/cloud-ai-platform/devdiv/serverless-paas-balam/serverless-paas-vikr/app-service-web-apps/app-service-team-documents/generalteamdocs/security/sdl/codeql)

Once the PR and justification look good, will send it to Anirudh for approval as per the guidance [here](https://strikecommunity.azurewebsites.net/articles/9993/continuous-sdl-cai-security-exceptions.html).

We calculate md5 for a blob before uploading a zip package for run from package. This is validated against the precalculated md5 from blob client on azure to ensure the full zip file got uploaded. We cannot change this as blob client provides md5 out of the box.

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: src/Azure.Functions.Cli/Helpers/SecurityHelpers.cs [hunk 1]
```diff
@@ -121,6 +121,10 @@ internal static async Task<X509Certificate2> CreateCertificateOpenSSL()
             return new X509Certificate2($"{certFileNames}certificate.pfx", DEFAULT_PASSWORD);
         }
 
+        // Suppressing CodeQL warning because this code is only used to validate blob content. Not used to encrypt any secrets
+        // blobClient only provides md5 string for validation. We cannot replace that unless they provide something else for validation
+        // TODO: Find out if 
+#pragma warning disable CodeQL.SM02196
```
Comment (jviau) [inline src/Azure.Functions.Cli/Helpers/SecurityHelpers.cs:127]: I don't believe this is the correct syntax to suppress CodeQL.

Comment (soninaren) [reply src/Azure.Functions.Cli/Helpers/SecurityHelpers.cs:127]: I got this from co-pilot 😊 was trying to see if I could check code ql status from pr build.

Comment (jviau) [reply src/Azure.Functions.Cli/Helpers/SecurityHelpers.cs:127]: No CodeQL is not ran in PR. It is ran once every 72 hours on CI (non-PR) builds

Comment (surgupta-msft) [reply src/Azure.Functions.Cli/Helpers/SecurityHelpers.cs:127]: @soninaren curious how did you get this opaque ID - SM02196?

Comment (soninaren) [reply src/Azure.Functions.Cli/Helpers/SecurityHelpers.cs]: fixed the syntax

File: src/Azure.Functions.Cli/Helpers/SecurityHelpers.cs [hunk 2]
```diff
@@ -121,6 +121,9 @@ internal static async Task<X509Certificate2> CreateCertificateOpenSSL()
             return new X509Certificate2($"{certFileNames}certificate.pfx", DEFAULT_PASSWORD);
         }
 
+
```
Comment (liliankasem) [inline src/Azure.Functions.Cli/Helpers/SecurityHelpers.cs:124]:
```suggestion
```

File: src/Azure.Functions.Cli/Helpers/SecurityHelpers.cs [hunk 3]
```diff
@@ -121,10 +121,9 @@ internal static async Task<X509Certificate2> CreateCertificateOpenSSL()
             return new X509Certificate2($"{certFileNames}certificate.pfx", DEFAULT_PASSWORD);
         }
 
-        // Suppressing CodeQL warning because this code is only used to validate blob content. Not used to encrypt any secrets
+
+        // CodeQL [SM02196] intent is to validate blob content. Not used to encrypt any secrets
```
Comment (jviau) [inline src/Azure.Functions.Cli/Helpers/SecurityHelpers.cs:125]: This is on the wrong line. It needs to be on same line as `using (var md5 = MD5.Create())` or be immediately above it.

``` CSharp
// CodeQL [SM02196] intent is to validate blob content. Not used to encrypt any secrets
using (var md5 = MD5.Create())
```

OR

``` CSharp
using (var md5 = MD5.Create()) // CodeQL [SM02196] intent is to validate blob content. Not used to encrypt any secrets
```

**Review by jviau — DISMISSED** [review]

**Review by liliankasem — APPROVED** [review]

**Review by jviau — APPROVED** [review]

**Review by anirudhgarg — APPROVED** [review]
Approved based on what is provided out of box by Blob client and this is not used for encryption purposes.

**Pull request merged**